## Info
```
Thanks Xin-
The file I shared can be found here:
T:\DCEG\CGF\Sequencing\ONT\Prom24\RawData\AS_chrX_ONT_training_07012025\SD386613\20250701_1804_1C_PBE55027_8e8920e8 (the only html file in this folder)

It is autogenerated at the end of the run.  The config settings are in the report itself. My guess is that the data it is using is:

T:\DCEG\CGF\Sequencing\ONT\Prom24\RawData\AS_chrX_ONT_training_07012025\SD386613\20250701_1804_1C_PBE55027_8e8920e8\bam_pass
 
Just remember that this is the data that was generated during live basecalling, and the data you are using is from a subsequent
analysis where SUP basecalling was performed across the entire dataset. What I was trying to highlight was simply that the preliminary
result of 20% of the reads shows a higher proportion of reads on chrX compared to other chromosomes, as well as ~150x coverage across
the X chromosome.  So it stands to reason that when using the data from the whole flowcell, we should STILL see a higher proportion of
reads on chrX, as well as substantially more than 150x on chromosome X. 
```
